### ISSUE TITLE
search-summary and search produce different results

### ISSUE THREAD

**Issue by C0nsultant** [eikek/docspell#1350]
Hi there

I recently switched DMS to docspell, really like it so far and plan to stick with it.
Because it will handle all of my paperwork, I want to have a backup option in addition to the raw database (I agree it is an easy way that works well, but when backing up important data I prefer two formats).

To get the data out of the database without having to write my own exporter, I wanted to take advantage of `dsc` .
But unfortunately it does not find/export all of my data. `search-summary` correctly displays the database contents:
```bash
$ dsc search-summary '!tag=ToDo'
All
┌───────┐
│ items │
├───────┤
│ 469   │
└───────┘
[...]
```
`search` (and consequently, `export`  and `download`) does not, however:
```bash
$ dsc search --all --limit 1000 '!tag=ToDo' | wc -l
204
```
Disregarding the header and footer lines of `dsc search`, this does ignore more than half of the documents.
When exporting/downloading using the same approach, only the above listed 200 documents actually appear.

The same behaviour can be reproduced with the raw API endpoints:
```bash
$ curl -X 'POST' 'http://127.0.0.1:7880/api/v1/sec/item/searchStats'  -d '{ "query": "!tag=ToDo" }' | jq .count
469
```
```bash
$ curl -X 'POST' 'http://127.0.0.1:7880/api/v1/sec/item/search' -d '{ "limit": 1000, "query": "!tag=ToDo" }' | jq | grep '"source"' | wc -l
200
```

For completeness: I am running the `latest` versions of the images in the official docker-compose stack.

**Comment by eikek** [comment]
Hi @C0nsultant ,

I think this hits the `max-item-page-size` setting in the config file. This is there so admins can control the amount of load that happens when searching. It's default is `200`. You can increase it or search in a loop with increasing the `--offset`. The `export` command has this implemented and will download all data available through the search interface. by doing multiple search requests. Do you see this with `export`, too?

This is not really documented, other than in the default settings [here](https://docspell.org/docs/configure/#rest-server), I'll add it to the api docs.

**Comment by C0nsultant** [comment]
Thank you for coming back at this so quickly! You are correct, indeed `max-item-page-size` is the culprit here. I increased it and as a result, the `/search` endpoint as well as `dsc search` now return more than 200 results. 

Having the option to supply a `--limit` argument is honestly a little confusing when the global hard-limit is used silently instead. May I suggest printing a log message or even returning a code 400 when `--limit` > `max-item-page-size`?

To your second question: Yes, `dsc export` does create more than 200 files (even without modification), but I'm running into a path-length issue there:
> Cmd: Export: Error creating a file: Filename too long (os error 36)

As a result of this error, I can not `export` the entire dataset (but there are more than 200 at least).

**Comment by eikek** [comment]
Yes, a 422 could be used here. Or it could be returned with the result so clients can print a message to the user. Thanks, I'll think about it.

Regarding the filename error: what filesystem are you using here? I think extfs and also ntfs should be ok wifh up to 255 characters. I think all names are capped to 254 characters at the database. Is there more to the error? You could try to run with `--verbose` maybe it gives a clue where this happens.

**Comment by C0nsultant** [comment]
Sorry, it took me  a while to dig deeper into the filename issue.

The filesystem in use is ZFS, which has a path length limit of 255 bytes/characters.
I did try to run with `--verbose`, but that did not display the [log::debug](https://github.com/docspell/dsc/blob/v0.7.0/src/cli/cmd/export.rs#L220) lines. After modifying with to `eprintln` instead, the faulty document is
> 26VJtaU8PV7-nBY1k6sZYQR-EWHpsbLEArj-4LJv5xcZ4Dw/XXXXXXX_XXXX_Unterjährige Entgeltaufstellung_vom_XX.XX.XXXX_XXXXXXXXXXXXXX

Which still did not explain the path length issue, as this should create a file 
> /home/XXXXXXXXXXX/tmp/dsc/foo/items/26/files/26VJtaU8PV7-nBY1k6sZYQR-EWHpsbLEArj-4LJv5xcZ4Dw/files/XXXXXXX_XXXX_Unterjährige Entgeltaufstellung_vom_XX.XX.XXXX_XXXXXXXXXXXXXX.pdf

with length 177. After debugging a little further, it turns out the [full file_path](https://github.com/docspell/dsc/blob/v0.7.0/src/cli/cmd/export.rs#L254) is 
> /home/XXXXXXXXXXX/tmp/dsc/foo/items/26/26VJtaU8PV7-nBY1k6sZYQR-EWHpsbLEArj-4LJv5xcZ4Dw/files/XXXX-XX-XX_Volksbank Raiffeisenbank_XXXXXXX_XXXX_Unterj?hrige Entgeltaufstellung_vom_XX.XX.XXXX_XXXXXXXXXXXXXX_Rechnung_ToDo.pdf"; filename*=UTF-8''XXXX-XX-XX_Volksbank%20Raiffeisenbank_XXXXXXX_XXXX_Unterj%C3%A4hrige%20Entgeltaufstellung_vom_XX.XX.XXXX_XXXXXXXXXXXXXX_Rechnung_ToDo.pdf

with length 376. I have no idea where the escaped `filename*` part comes from as the document is correctly tagged and shows no such bugs in the web ui or ` dsc search`:
```bash
& dsc search "name:*Unterjährige*"
┌──────────┬────────────────────────────────────────────────────────────────────────────┬───────────┬────────────┬─────┬──────────────────────────┬────────────┬────────┬──────┬────────┬───────┐
│ id       │ name                                                                       │ state     │ date       │ due │ correspondent            │ concerning │ folder │ tags │ fields │ files │
├──────────┼────────────────────────────────────────────────────────────────────────────┼───────────┼────────────┼─────┼──────────────────────────┼────────────┼────────┼──────┼────────┼───────┤
│ 26VJtaU8 │ XXXXXXX_XXXX_Unterjährige Entgeltaufstellung_vom_XX.XX.XXXX_XXXXXXXXXXXXXX │ confirmed │ XXXX-XX-XX │     │ Volksbank Raiffeisenbank │            │        │ ToDo │        │ 1     │
└──────────┴────────────────────────────────────────────────────────────────────────────┴───────────┴────────────┴─────┴──────────────────────────┴────────────┴────────┴──────┴────────┴───────┘
```

**Comment by eikek** [comment]
Thanks a lot for the update! To enable debug statements, I think you need to specify more than one verbose, like in `-vv` - sorry for not being mentioned in the help text :-| 

This is really strange behavior. Could you maybe check the original filenames in your docspell database?  For example with this select:
```sql
select filename from attachment_source where filename like '%;%';
```
the `filename` column in the table `attachment_source` is the original filename as it was uploaded. It would be interesting to rule out (or not) the server side.

Then a guess is that this affects names with umlauts? Then for some reason this `filename*=` is appended and doubles the filename length.

**Comment by eikek** [comment]
Just tested it with your filename `XXXXXXX_XXXX_Unterjährige Entgeltaufstellung_vom_XX.XX.XXXX_XXXXXXXXXXXXXX.pdf` - and could reproduce it! Very strange! So it seems to be a bug in `dsc` and not docspell server. I was sure I tested this with umlauts, but maybe something crept in.

**Comment by C0nsultant** [comment]
> To enable debug statements, I think you need to specify more than one verbose, like in -vv

Should have thought of that before modifying source code...

> Could you maybe check the original filenames in your docspell database?

You have identified the problem already, but just for completeness' sake:
The raw database query also shows the correct filename `XXXXXXX_XXXX_Unterjährige Entgeltaufstellung_vom_XX.XX.XXXX_XXXXXXXXXXXXXX.pdf` (the `X`s are just numbers I redacted).

I just tried `0.7.1`/`0.8.0-pre` and in both cases the bug does not occur. All documents are exported successfully. Thank you so much, @eikek :)
